### PR TITLE
Add support for Wikimedia's PAWS instance

### PR DESCRIPTION
**Pull request by yuvipanda**
1. MediaWiki Authenticator support
2. Making singleuser uid and gid configurable
3. Stop running hub process as root! 
4. Support adding arbitrary extra volumes and mounts to user pod

### CODE REVIEW

File: images/hub/jupyterhub_config.py [hunk 1]
```diff
@@ -51,9 +51,13 @@ def get_config(key, default=None):
 
 c.KubeSpawner.singleuser_extra_labels = get_config('singleuser.extra-labels', {})
 
-# FIXME: Make this better? 
-c.KubeSpawner.singleuser_uid = 1000
-c.KubeSpawner.singleuser_fs_gid = 1000
+c.KubeSpawner.singleuser_uid = get_config('singleuser.uid')
+c.KubeSpawner.singleuser_fs_gid = get_config('singleuser.fs-gid')
+
+# FIXME: Make this the default? This lets us do + with lists for
```
Comment (choldgraf) [inline images/hub/jupyterhub_config.py]: seems reasonable to me

Comment (minrk) [inline images/hub/jupyterhub_config.py]: This is fine, though you can use `.extend` on config values without initializing them and it figures out that it's a list.

File: images/hub/jupyterhub_config.py [hunk 2]
```diff
@@ -109,6 +98,8 @@ def get_config(key, default=None):
         'subPath': get_config('singleuser.storage.static.sub-path')
     }]
 
+c.KubeSpawner.volumes = c.KubeSpawner.volumes + get_config('singleuser.storage.extra-volumes')
```
Comment (minrk) [inline images/hub/jupyterhub_config.py]: Probably better to use `volumes.extend(get_config...)`

**Comment by yuvipanda** [comment]
I'm merging this into the jupyterhub-0.8 branch as part of larger work! Thanks for the review everyone :)

**Pull request closed without merging**